initial conversion of main services - build and deploy

matthewgardner opened the pull request:
THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE FINOS Corporate Contributor License Agreement.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

https://github.com/finos/traderX/issues/203

Complete
- Added Tilt file for local execution
- Added Base k8s config
- Added Yaml for most services
- Simple Port Ingress (will revisit in another PR)

DovOps commented:
@matthewgardner  once we sort out the CLA situation, we're ready to merge this. I think you can do this by clicking the red CLA button above.

bingenito commented:
/easycla

matthewgardner commented:
- Added Nginx Ingress
- Added to readme for how to add to local cluster

<img width="1290" alt="Screenshot 2024-09-10 at 21 02 29" src="https://github.com/user-attachments/assets/2a5220ea-97f7-4f4d-b349-a48991c89bdc">

Suggestions for future pr (out of scope of this PR?) is to enable health endpoints on service so make it easy to spot when services are actually ready.

matthewgardner commented on web-front-end/angular/Dockerfile:
```diff
@@ -17,4 +17,4 @@ RUN su node -c "source /usr/local/share/nvm/nvm.sh && nvm install ${EXTRA_NODE_V
 RUN npm install --ignore-scripts puppeteer
-ENTRYPOINT npm run start-prod
\ No newline at end of file
+ENTRYPOINT npm run ${WEB_FRONT_END_START_MODE:-start-prod}
```
@DovOps  should i revert this change?

DovOps replied:
Yeah, if ingress is working you don't need to connect to each service via its port, but rather connect only via the ingress controller, but test to make sure the app works fine that way.

matthewgardner replied:
Done - also tested on a windows env :)

DovOps commented:
> 
> Suggestions for future pr (out of scope of this PR?) is to enable health endpoints on service so make it easy to spot when services are actually ready.

Definitely. See issue #124

DovOps commented:
/easycla

DovOps commented:
@bingenito  @maoo  @karlmoll  - Need a bit of help sorting out the easyCLA issue, as it has to do with a commit from me that is not using the right e-mail.  Tried fixing this but it did not work. Once done we can merge this, unless we should merge and note the exception.

maoo commented:
> @bingenito @maoo @karlmoll - Need a bit of help sorting out the easyCLA issue, as it has to do with a commit from me that is not using the right e-mail. Tried fixing this but it did not work. Once done we can merge this, unless we should merge and note the exception.

@DovOps , could you please confirm that the email address that you see on https://api.github.com/repos/finos/traderX/commits/cd9ffec6febfe75913dcf95e26e0b1682e75c79f is also on https://github.com/settings/emails ?

In parallel, I'd also check `git config --list | grep email` to make sure which email you use to sign commits; another - maybe longer - option is to update the `user.email` config, `git reset` and then commit again on @matthewgardner branch.

DovOps commented:
@maoo  it's the noreply one for me, which apparently was in my settings on the local repo for some reason.  I'll try a git reset + commit if that works

DovOps commented:
This did not succeed. Merging this as-is.  @bingenito  can confirm that @DovOps  (author of https://github.com/finos/traderX/commit/cd9ffec6febfe75913dcf95e26e0b1682e75c79f ) has agreed to CLA.

bingenito commented:
> This did not succeed. Merging this as-is. @bingenito can confirm that @DovOps (author of [cd9ffec](https://github.com/finos/traderX/commit/cd9ffec6febfe75913dcf95e26e0b1682e75c79f) ) has agreed to CLA.

As CLA Manager for Morgan Stanley I confirm that @DovOps is covered under our EasyCLA
